src/main/groovy/com/ullink/gradle/opencover/OpenCover.groovy

```diff
@@ -80,10 +104,64 @@ class OpenCover extends ConventionTask {
         if (skipAutoProps) commandLineArgs += '-skipautoprops'
         if (hideSkipped) commandLineArgs += '-hideskipped:' + hideSkipped
 
+        commandLineArgs
+    }
+
+    def runSingleOpenCover(ArrayList commandLineArgs) {
         commandLineArgs += ["-target:${getTargetExec()}", "\"-targetargs:${getTargetExecArgs().collect({escapeArg(it)}).join(' ')}\"", "-targetdir:${project.buildDir}"]
         def filters = getTargetAssemblies().collect { "+[${FilenameUtils.getBaseName(project.file(it).name)}]*" }
         commandLineArgs += '-filter:\\"' + filters.join(' ') + '\\"'
         commandLineArgs += "-output:${getCoverageReportPath()}"
+
+        execute(commandLineArgs)
+    }
+
+    def runMultipleOpenCovers(ArrayList commandLineArgs) {
+        def intermediateReportsPath = new File(reportsFolder, "intermediate-results-" + name)
+        intermediateReportsPath.mkdirs()
+
+        def runs = project.nunit.getTestInputAsList(getWhereFilters())
+
+        GParsPool.withPool {
+            runs.eachParallel {
+                def fileName = getRandomFileName()
+                logger.info("Filename generated for the \'$it\' input was \'$fileName\'")
+
+                def execArgs = project.nunit.buildCommandArgs(it, new File(intermediateReportsPath, fileName + ".xml"))
+                commandLineArgs += ["-target:${getTargetExec()}", "\"-targetargs:${execArgs.collect({escapeArg(it)}).join(' ')}\"", "-targetdir:${project.buildDir}"]
+                commandLineArgs += "-output:${new File(intermediateReportsPath, fileName + ".xml")}"
+
+                execute(commandLineArgs)
+            }
+        }
+
+        execute(getMergeCommand(intermediateReportsPath))
+
+        new File(reportsFolder, "Summary.xml").renameTo(new File(reportsFolder, "coverage.xml"))
+        new File(intermediateReportsPath, "").deleteDir()
+    }
+
+    def getRandomFileName() {
+          UUID.randomUUID().toString()
+      }
+
+    def shouldRunInParallel() {
+        if (!project.nunit.parallelForks || getWhereFilters() == null)
```

Comment:
I think we should throw an exception that where filters are not specified when trying to run these as paralel.
keep this method just `return project.nunit.parallelForks` and where it's used add
`assert getWhereFilters(), "You must specify where filters by which to parallelize the run`

Reply:
Not quite the case here. I tried that as a first option and didn't worked out well: in case we don't need to have parallelization the parallelForks is by default true, so even if we need a single run it will try to get the where filter (which is not needed in this case) and even if the hasProperty('where') will return true, nunit.where is null => it will crash when we do nunit.where.value. That's why we need both conditions, in order to make sure we don't actually break the previous working mechanism.